src/helpers/ConfigHelper.ts

```diff
@@ -230,26 +299,71 @@ export class ConfigHelper {
       }
     };
 
+    let defaultSlidedownOptions;
+
     if (promptOptionsConfig.slidedown) {
-      promptOptionsConfig.slidedown.enabled = !!promptOptionsConfig.slidedown.enabled;
-      promptOptionsConfig.slidedown.autoPrompt = promptOptionsConfig.slidedown.hasOwnProperty("autoPrompt") ?
-        !!promptOptionsConfig.slidedown.enabled && !!promptOptionsConfig.slidedown.autoPrompt :
-        !!promptOptionsConfig.slidedown.enabled;
-        promptOptionsConfig.slidedown.pageViews = Utils.getValueOrDefault(promptOptionsConfig.slidedown.pageViews,
-          SERVER_CONFIG_DEFAULTS_PROMPT_DELAYS.pageViews);
-        promptOptionsConfig.slidedown.timeDelay = Utils.getValueOrDefault(promptOptionsConfig.slidedown.timeDelay,
-          SERVER_CONFIG_DEFAULTS_PROMPT_DELAYS.timeDelay);
-
-      if (promptOptionsConfig.slidedown.categories) {
-        const { categories } = promptOptionsConfig.slidedown;
-        promptOptionsConfig.slidedown.categories = TagUtils.limitCategoriesToMaxCount(categories, MAX_CATEGORIES);
-      }
+      promptOptionsConfig.slidedown.prompts = promptOptionsConfig.slidedown?.prompts?.map(promptOption => {
+        promptOption.type = Utils.getValueOrDefault(promptOption.type, DelayedPromptType.Push);
+
+        if (promptOption.type === DelayedPromptType.Category) {
+          promptOption.text = {
+            ...promptOption.text,
+            positiveUpdateButton: Utils.getValueOrDefault(promptOption.text.positiveUpdateButton,
+              SERVER_CONFIG_DEFAULTS_SLIDEDOWN.categoryDefaults.positiveUpdateButton),
+            negativeUpdateButton: Utils.getValueOrDefault(promptOption.text.negativeUpdateButton,
+              SERVER_CONFIG_DEFAULTS_SLIDEDOWN.categoryDefaults.negativeUpdateButton),
+            updateMessage: Utils.getValueOrDefault(promptOption.text.updateMessage,
+              SERVER_CONFIG_DEFAULTS_SLIDEDOWN.categoryDefaults.updateMessage),
+          };
+        }
+
+        promptOption.text = {
+          ...promptOption.text,
+          actionMessage: Utils.getValueOrDefault(promptOption?.text?.actionMessage,
+            SERVER_CONFIG_DEFAULTS_SLIDEDOWN.actionMessage),
+          acceptButton: Utils.getValueOrDefault(promptOption?.text?.acceptButton,
+            SERVER_CONFIG_DEFAULTS_SLIDEDOWN.acceptButton),
+          cancelButton: Utils.getValueOrDefault(promptOption?.text?.cancelButton,
+            SERVER_CONFIG_DEFAULTS_SLIDEDOWN.cancelButton)
+        };
+
+        promptOption.autoPrompt = Utils.getValueOrDefault(promptOption.autoPrompt, false);
```

Comment:
switched the default to true. looking at the [current logic](https://github.com/OneSignal/OneSignal-Website-SDK/blob/master/src/helpers/ConfigHelper.ts#L235-L236) it appears that we set `autoPrompt` to the configured value if it is set. otherwise, we default to the `enabled` value. But in this case, `enabled` is deprecated so if the slidedown option exists, it means we want it to be enabled (i.e. = true)